### PR TITLE
verify_rule fix for negated groups

### PR DESCRIPTION
**Pull request by marcbou**
## Change Summary
verify_rule fix for negated groups (name prefixed by '!')

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/Txxxx

## Checklist:

- [ ] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [ ] I have linked this PR to one or more Phabricator Task(s)
- [ ] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [ ] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/conf_mode/policy-route.py [hunk 1]
```diff
@@ -123,6 +123,10 @@ def verify_rule(policy, name, rule_conf, ipv6):
                 for group in valid_groups:
                     if group in side_conf['group']:
                         group_name = side_conf['group'][group]
+
+                        if group_name and group_name[0] == '!':
```
Comment (c-po) [inline src/conf_mode/policy-route.py:127]: What about geoup_name.startswith()?

**Comment by sever-sever** [comment]
Could you set a valid task id in the commit?
https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md

File: src/conf_mode/policy-route.py (hunk 1 above)
Comment (marcbou) [reply src/conf_mode/policy-route.py:127]: the style mimics similar existing code in src/conf_mode/firewall.py:verify_rule()

**Comment by c-po** [comment]
@sarthurdev what you think?

**Comment by sarthurdev** [comment]
> @sarthurdev what you think?

Looks good to me

File: src/conf_mode/policy-route.py (hunk 1 above)
Comment (dmbaturin) [reply src/conf_mode/policy-route.py:127]: Existing code isn't always good code. ;)
I'm also for `startswith`.

**Review by dmbaturin — CHANGES_REQUESTED** [review]

File: src/conf_mode/policy-route.py (hunk 1 above)
Comment (marcbou) [reply src/conf_mode/policy-route.py:127]: how many github "reviewers" does it take to change a lightbulb ?

it's trivial to fix, whether in existing code style or with startswith() or whatever.. but negated group support is important, and this bug breaks it so it should be fixed in whichever way or style you prefer !

**Pull request closed without merging**